### DISCUSSION TITLE
autocomplete popular websites by a single word

### DISCUSSION THREAD

**Discussion by shalva97** [minbrowser/min#1714] [Ideas]
These days I have been thinking about autocompletion. Lets say I want to go to Github, this is what I get if I type just Github
![image](https://user-images.githubusercontent.com/22417494/129457190-3efaf0af-7547-4fcb-a969-b0275ee28e61.png)

I dont need any of them, maybe its just me but idk... all that is useless for me. the first three is fine, but the last 4 look extra. 

Anyways, I have noticed a pattern of my habits. lets say I want to go to Github and it is not yet saved by Min, so it doesnot suggest me a github.com. But then, 100% of times I just press enter and search it on google, because of my lazynes, I rarely remember .com or .net or .org.

so my idea is to suggest popular websites if user types a single word. If user types one word and then space then it would show common suggestions that is in the pic.

**Comment by PalmerAL** [comment]
Yeah, this is a pretty good idea. It wouldn't be too hard to bundle a list of the top 100 or so websites, and show them as history suggestions along with pages you've visited. I think we wouldn't need the special "then show suggestions on space behavior" - the top site URL could just go at the top of the list and be autocompleted.

**Reply by lubiedo** [reply]
We can also reuse the [Alexa top 1000 sites list](https://github.com/minbrowser/min/blob/master/ext/httpsUpgrade/httpsTopSites.json) that we already use for the HTTPS upgrade.

**Comment by PalmerAL** [comment]
I realized that DDG already provides something like this in their API, but we weren't using it effectively (hiding it most of the time, and displaying it at the bottom of the list when we did show it). I made a branch that improves this: https://github.com/minbrowser/min/tree/top-site-ac. If you type most business names ("microsoft", "instagram"), it will display the site at the top now and autocomplete the URL.

It has a couple limitations compared to using a built-in top site list:
* DDG is missing official sites for some domains (github and twitter don't work so far)
* You have to type the whole name for it to autocomplete (eg you have to type out "microsoft" and then it autocompletes ".com")
* It won't work if you pick a search engine other than DDG.
